src/wavm/openmp.cpp

```diff
@@ -1,60 +1,37 @@
-#include "wavm/openmp/openmp.h"
-
+#include "faabric/util/gids.h"
 #include <WAVM/Platform/Thread.h>
 #include <WAVM/Runtime/Intrinsics.h>
 #include <WAVM/Runtime/Runtime.h>
-#include <future>
 
 #include <faabric/scheduler/Scheduler.h>
 #include <faabric/state/StateKeyValue.h>
+#include <faabric/util/locks.h>
 #include <faabric/util/timing.h>
-#include <wavm/OMPThreadPool.h>
+
+#include <wavm/ThreadState.h>
 #include <wavm/WAVMWasmModule.h>
-#include <wavm/openmp/Level.h>
-#include <wavm/openmp/ThreadState.h>
 
 using namespace WAVM;
 
 namespace wasm {
-using namespace openmp;
 
-/**
- * Performs actual static assignment
- */
-template<typename T>
-void for_static_init(I32 schedule,
-                     I32* lastIter,
-                     T* lower,
-                     T* upper,
-                     T* stride,
-                     T incr,
-                     T chunk);
+// ------------------------------------------------
+// LOGGING
+// ------------------------------------------------
 
-/**
- * Function used to spawn OMP threads. Will be called from within a thread
- * (hence needs to set up its own TLS)
- */
-I64 ompThreadEntryFunc(void* threadArgsPtr)
-{
-    auto args = *reinterpret_cast<LocalThreadArgs*>(threadArgsPtr);
-    // Set up various TLS
-    setTLS(args.tid, args.level);
-    setExecutingModule(args.parentModule);
-    setExecutingCall(args.parentCall);
-    return getExecutingWAVMModule()->executeThreadLocally(args.spec);
-}
+#define OMP_FUNC(str)                                                          \
+    auto ctx = getOpenMPContext();                                             \
+    auto logger = faabric::util::getLogger();                                  \
+    logger->debug("OMP {}: " str, ctx.threadNumber);
```

Comment:
Added some convenience macros

Reply:
If it's cumbersome to change it's not a big deal, but with the changes we made to loggers in faasm/faabric#56, you could do something like:
```cpp
auto logger = faabric::util::getLogger(fmt::format("OMP-{}", ctx.threadNumber));
logger->debug(str);
```